src/hotspot/cpu/x86/stubGenerator_x86_64_fmod.cpp

```diff
@@ -319,6 +314,12 @@ address StubGenerator::generate_libmFmod() {
 
     //   if (((mxcsr & 0x6000)!=0x2000) && (a < b * 0x1p+260))
     __ bind(L_104a);
+    __ stmxcsr(Address(rsp, 0));
+    __ movl(rax, Address(rsp, 0));
+    __ movl(rcx, rax);
+    __ orl(rcx, 0x7f80);
+    __ movl(Address(rsp, 0x04), rcx);
+
     __ andl(rax, 0x6000);
     __ cmpl(rax, 0x2000);
     __ jcc(Assembler::equal, L_10c1);
```

Comment:
Some of the jcc and jmp could be generated as short branches where feasible.
Also the stub and the loop entries could be aligned to 16 bytes for better perf.